src/main/scala/mesosphere/marathon/api/v2/json/V2AppUpdate.scala

```diff
@@ -119,11 +124,31 @@ case class V2AppUpdate(
 object V2AppUpdate {
   import com.wix.accord.dsl._
   import mesosphere.marathon.api.v2.Validation._
+
+  private def updateContainsEitherUrisOrFetch: Validator[V2AppUpdate] = {
+    new Validator[V2AppUpdate] {
+      def apply(app: V2AppUpdate): Result = {
+
+        val fetch = app.fetch.getOrElse(AppDefinition.DefaultFetch)
+        val uris = app.uris.getOrElse(AppDefinition.DefaultUris)
+
+        if (fetch.nonEmpty && uris.nonEmpty) {
```

Comment:
app.fetch and app.uris is an Option, so you can do:
```scala
if (app.fetch.nonEmpty && app.uris.nonEmpty) 
```
without mapping into the option.